src/utils/device.js

```diff
@@ -0,0 +1,69 @@
+/**
+* Returns an object with detected device and browser.
```

Comment:
I'm not sure we need this all browser device detection code for iOS.
It could be as simple as:
```js
    // iOS detection from: http://stackoverflow.com/a/9039885/177710
    if (/iPad|iPhone|iPod/.test(userAgent) && !window.MSStream) {
        return "iOS";
    }
```